OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -570,7 +570,9 @@ public Activity MoveIntoWorld(Actor self, CPos cell, SubCell subCell = SubCell.A
 			SetPosition(self, cell, subCell);
 			SetVisualPosition(self, pos);
 
-			return WrapMove(VisualMove(self, pos, self.World.Map.CenterOfSubCell(cell, subCell), cell));
+			var activities = new Wait(1, false);
```

Comment:
The call to `SetVisualPosition` cannot happen in the same frame as the `Drag` activity. This is what caused the glitch, probably because the `ActorMap` isn't properly updated yet. So we need to delay the activity for one frame.

Moving the call to `MoveIntoWorld` out of the `AddFrameEndTask` block compensates for the delay.

Reply:
Could possible be the cause of this change https://github.com/OpenRA/OpenRA/pull/16698/files#diff-ae5e31bf8fcc90fa4056261f07135868L518 , which i was supposed to reset...

Reply:
That is probably it. If you can fix the underlying issue as part of #16861 then this PR should probably be closed.

Reply:
I can't reproduce #16901 on the current version of #16861 so it seems the issue is already fixed, from something else, there.